bd-client-stats/src/stats.rs

```diff
@@ -226,14 +243,19 @@ impl<F: SerializedFileSystem> Uploader<F> {
       return Ok(());
     }
 
-    let Ok(aggregated_stats) = self
+    let aggregated_stats = match self
       .fs
       .read_compressed_protobuf_file::<StatsSnapshot>(&*AGGREGATED_STATS_FILE)
       .await
-    else {
-      log::warn!("failed to read aggregated stats file due to data corruption, deleting file");
-      let _ignored = self.fs.delete_file(&*AGGREGATED_STATS_FILE).await;
-      return Ok(());
+    {
+      Ok(stats) => stats,
+      Err(e) => {
+        log::warn!(
+          "failed to read aggregated stats file due to data corruption, deleting file: {e}"
+        );
+        let _ignored = self.fs.delete_file(&*AGGREGATED_STATS_FILE).await;
```

Comment:
Don't remember why we ignore the error here but not in the other delete_file call but seems a bit strange

Reply:
I'm not sure reporting failure to delete is useful so maybe I will just remove the other one.

Reply:
In looking at this the error handling here is all over the place. Let me try to clean it up.